src/garage/torch/baselines/gaussian_mlp_baseline.py

```diff
@@ -0,0 +1,170 @@
+        min_std (float): If not None, the std is at least the value of min_std,
+            to avoid numerical issues (plain value - not log or exponentiated).
+        max_std (float): If not None, the std is at most the value of max_std,
+            to avoid numerical issues (plain value - not log or exponentiated).
+        std_parameterization (str): How the std should be parametrized. There
+            are two options:
+            - exp: the logarithm of the std will be stored, and applied a
+               exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+        normalize_observation (bool): Bool for normalizing observation or not.
+        normalize_return (bool): Bool for normalizing return or not.
+        optimizer (object): The optimizer of the algorithm. Should be the
+            optimizers in torch.optim.
+        optimizer_args (dict): Optimizer arguments.
+    """
+
+    def __init__(self,
+                 env_spec,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=torch.tanh,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 learn_std=True,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_parameterization='exp',
+                 layer_normalization=False,
+                 normalize_observation=True,
+                 normalize_return=True,
+                 optimizer=torch.optim.Adam,
+                 optimizer_args=None):
+        super().__init__(env_spec)
+
+        self._module = GaussianMLPModule(
+            input_dim=env_spec.observation_space.flat_dim,
+            output_dim=1,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            learn_std=learn_std,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization)
+
+        self._normalize_observation = normalize_observation
+        self._normalize_return = normalize_return
+
+        if optimizer_args is None:
+            optimizer_args = dict()
+
+        self._optimizer = make_optimizer(optimizer, self._module,
+                                         **optimizer_args)
+
+    def fit(self, paths):
+        """Fit regressor based on paths.
+
+        Args:
+            paths (list[dict]): Sample paths.
```

Comment:
please tell us a lot more about what needs to be in this data structure. I recommend  you use TrajectoryBatch if you can to avoid writing a lot of documentation.

your signature could be
```python
def fit(self, trajs, returns):
    r"""Update the baseline using these trajectories and returns.
    
    Args:
        trajs (TrajectoryBatch): Trajectories to fit with.
        returns (np.ndarray): Returns labels for the trajectories in :obj:`traj`, with shape :math:`(N \bullet [T],)`
    """

Reply:
`TrajectoryBatch` doesn't have `return` values so I can't use the class.

Reply:
can you read my example above? i pass returns as a separate argument. it doesn't have to be all-or-nothing.